python/tutorials/06-fused-attention.py

```diff
@@ -349,10 +349,19 @@ def test_op(Z, H, N_CTX, D_HEAD, causal, dtype=torch.float16):
     sm_scale = 0.5
     dout = torch.randn_like(q)
     # reference implementation
-    M = torch.tril(torch.ones((N_CTX, N_CTX), device="cuda"))
-    p = torch.matmul(q, k.transpose(2, 3)) * sm_scale
     if causal:
-        p[:, :, M == 0] = float("-inf")
+        M = torch.triu(
+            torch.full(
+                # Cannot give dtype here due to BF16 incompatibility with torch<=2.0:
+                # RuntimeError: "triu_tril_cuda_template" not implemented for 'BFloat16'
+                (N_CTX, N_CTX), float('-inf'), device="cuda"),
+            diagonal=1,
+        ).to(dtype)
+    else:
+        M = torch.zeros((N_CTX, N_CTX), device="cuda", dtype=dtype)
+
+    p = torch.matmul(q, k.transpose(2, 3)) * sm_scale
+    p = p + M
```

Comment:
what is the motivation for modifying the tutorial?

Reply:
I thought it would be nice to have it consistent between the two tests. If you disagree, I can revert the change.